File: uniffi/src/message/groups/attached_material_quadlets.rs
use cesrox::error::ParsideResult;
use cesrox::Group;
use cesrox::CesrGroup;
pub use cesrox::message::groups::AttachedMaterialQuadlets;

pub fn attached_material_quadlets_create(value: Vec<CesrGroup>) -> AttachedMaterialQuadlets {
    AttachedMaterialQuadlets::new(value)
}

pub fn attached_material_quadlets_qb64(attached_material_quadlets: &AttachedMaterialQuadlets) -> ParsideResult<String> {
    attached_material_quadlets.qb64()
}

pub fn attached_material_quadlets_qb64b(attached_material_quadlets: &AttachedMaterialQuadlets) -> ParsideResult<Vec<u8>> {
    attached_material_quadlets.qb64b()
}

pub fn attached_material_quadlets_qb2(attached_material_quadlets: &AttachedMaterialQuadlets) -> ParsideResult<Vec<u8>> {
    attached_material_quadlets.qb2()
}

File: wrappers/kotlin/lib/src/main/kotlin/cesrox/groups/AttachedMaterialQuadlets.kt
package cesrox.groups

import cesrox.CesrGroup
import uniffi.cesrox.*
import uniffi.cesrox.AttachedMaterialQuadlets as AttachedMaterialQuadletsNative

class AttachedMaterialQuadlets {
    val value: AttachedMaterialQuadletsNative

    constructor(couples: List<CesrGroup>) {
        value = attachedMaterialQuadletsCreate(couples.map { it.value })
    }

    constructor(couples: AttachedMaterialQuadletsNative) {
        value = couples
    }

    fun get(index: Int): CesrGroup {
        return CesrGroup(value.value[index])
    }

    fun qb2(): List<UByte> {
        return attachedMaterialQuadletsQb2(value)
    }
    fun qb64b(): List<UByte> {
        return attachedMaterialQuadletsQb64b(value)
    }
    fun qb64(): String {
        return attachedMaterialQuadletsQb64(value)
    }
}